components/Map/Map.jsx

```diff
@@ -1,167 +1,154 @@
-      )
-        .then((response) => response.text())
-        .then((csv) => {
-          const parsed = Papa.parse(csv, { header: true, dynamicTyping: true });
-          return Promise.resolve(parsed.data);
-        }),
-      fetch(
-        "https://raw.githubusercontent.com/bhrutledge/ma-legislature/main/dist/ma_house.geojson"
-      ).then((response) => response.json()),
-    ]).then(([supporters, houseFeatures]) => {
-      const supportersByDistrict = supporters.reduce((acc, cur) => {
-        acc[cur.district] = cur;
-        return acc;
-      }, {});
-
-      function repProperties(feature) {
-        const supporter = supportersByDistrict[feature.properties.district];
-        return supporter;
-      }
-
-      const houseLayer = L.geoJson(houseFeatures, {
-        style: (feature) => districtStyle(repProperties(feature)),
-        onEachFeature: (feature, layer) => {
-          const rep = repProperties(feature);
-          layer.bindPopup(districtPopup(rep));
-          layer.on("popupopen", onPopup);
-          layer.on("popupclose", onPopup);
-          feature.properties.index = `${rep.first_name} ${rep.last_name} - ${rep.district}`;
-        },
-      });
+    /* Load the legislative district boundaries and rep data */
 
-      const searchControl = new L.Control.Search({
-        layer: houseLayer,
-        propertyName: "index",
-        initial: false,
-        marker: false,
-        textPlaceholder: "Search reps and districts",
-        moveToLocation(latlng, title, map) {
-          map.fitBounds(latlng.layer.getBounds());
-          latlng.layer.openPopup();
-        },
+    Promise.all([
+      /* The GeoJSON contains basic contact information for each rep */
+      fetch('https://bhrutledge.com/ma-legislature/dist/ma_house.geojson').then((response) => response.json()),
+      fetch('https://bhrutledge.com/ma-legislature/dist/ma_senate.geojson').then((response) => response.json()),
+      /**
+       * To add additional data about each rep/district, uncomment this `fetch`
+       * and set the URL to a JSON data source with a `district` field.
+       *
+       * For a CSV, use https://www.papaparse.com/; see an example at:
+       * https://github.com/actonmass/campaign-map/blob/main/index.html
+       */
+      // fetch('https://example.com/rep_data.json').then((response) => response.json()),
+    ])
+      .then(([houseFeatures, senateFeatures, repData = []]) => {
+        /* Build a rep info object, e.g. `rep.first_name`, `rep.extra_data` */
+
+        const repDataByDistrict = repData.reduce((acc, cur) => {
+          acc[cur.district] = cur;
+          return acc;
+        }, {});
+
+        const repProperties = (feature) => {
+          const data = repDataByDistrict[feature.properties.district] || {};
+          return { ...feature.properties, ...data };
+        };
+
+        /* Templates for map elements */
+
+        const districtLegend = () => /* html */`
```

Comment:
I opted to keep all of the template functions within this scope because I think it's easier to follow.